doc/Variables.md

```diff
@@ -0,0 +1,117 @@
+# Variables
+
+A Variable _type_ represents some physical or logical property that is unknown or uncertain, along with any additional
+properties that are required to uniquely identify Variable instances from each other. Thus, every Variable type does
+two things. First, it holds the variable's _value_, acting as a one-or-more-dimensional tuple of floating-point data
+that defines the physical property. Second, it defines the variable's _identity_, holding the additional information
+that distinguishes one variable instance from another. Interestingly, and somewhat counter-intuitively, the _value_
+of the variable tuple **does not affect** the identity of a variable instance. The purpose of fuse is to solve for
+the most likely _value_ of all of the variables, and hence the variable instance _values_ will change over time as
+new evidence is accumulated. But the _identity_ of a variable instance is immutable, and independent of its _value_.
+
+A concrete example will help explain these concepts. We have two ground robots, cleverly named R1 and R2, and we want
+to estimate the pose of both robots as they drive around. Thus, the physical property of interest is the 2D pose. A
+2D pose consists of three floating-point quantities: `x`, `y`, and `'yaw`. Thus our Variable type must include a
+3-tuple to hold the variable's _value_. Now, what about the _identity_ portion of our Variable type? Well, we have two
+different robots, so we need to differentiate between the pose of R1 and the pose of R2. Thus our Variable type must
+include some sort of `robot_id` field. Different instances of our variable will set the `robot_id` field to either 'R1'
+or 'R2'. Additionally, the robots move around over time. So the pose of a robot is not valid for _any_ time, but
+represents the pose of the robot at a _specific_ time. And so our Variable type must also include a `timestamp` field
+to differentiate a pose at time `t1` for a pose at time `t2`.
+
+Just to reiterate the important concepts:
+
+* A variable type defines a _value_ that may be one or more floating-point quantities.
+* A variable instance's _value_ is unknown and/or uncertain. It will change over time as evidence is added.
```

Comment:
So I get what you're saying here: a given node in the graph, representing a variable, will have its value change as more variables and constraints are added to the graph over time. However, I can see someone confusing the notion of "variable value changing over time" and the notion of "a variable measuring pose at time t1 is a different variable than the measurement of pose at time t2." You call that fact out in the paragraph above, but it gets wrapped into the point on line 26 down here.

In EKF land, I have been referring to each dimension in the state as a variable, and its value changes over time. But we only have one state, so the value of, e.g., variable `x` is 23.1 at time `t1`, and 23.4 at time `t2`. But that's not the same as it is in graph land, where a variable is a measurement of N dimensions from a given sensor at a given time.